pkg/podimpersonation/podimpersonation.go

```diff
@@ -131,7 +131,7 @@ type PodOptions struct {
 //      namespace used only for the purpose of running impersonated pods. This is to ensure
 //      proper protection for the service accounts created.
 //   2. The pod must KUBECONFIG env var set to where you expect the kubeconfig to reside
-func (s *PodImpersonation) CreatePod(ctx context.Context, user user.Info, pod *v1.Pod, podOptions *PodOptions) (*v1.Pod, error) {
+func (s *PodImpersonation) CreatePod(ctx context.Context, user user.Info, pod *v1.Pod, podOptions *PodOptions, registryOverride string) (*v1.Pod, error) {
```

Comment:
I think a better approach would be to just add ImageName as a field on PodOptions that if set it will use that instead of calling the `func() string`.